hub/test/src/testDrivers/InMemoryDriver.js

```diff
@@ -0,0 +1,65 @@
+/* @flow */
```

Comment:
Is this driver used anywhere else besides the unit test?  If the only point to having this driver is to test the Gaia hub, then why have it at all?  Why not use the disk driver for testing Gaia?  If this is only for testing, then why have the in-memory driver in the main source tree, instead of within the test itself?

Reply:
This started off as extensions to the existing `MockDriver` and eventually I moved it into its own class. There are a couple things this driver does only for the purpose of testing e2e components (like auth token revocation), such as providing its own http read-end point. The revocation touches both driver code and authentication, which means that it needs to have a decent mock driver that can cache a value. I think using the disk driver would involve injecting/wrapping much of this test code around it.

Reply:
I think that the point is to be able to use it as a mock driver in unit tests --- using specific drivers for that is a pain, and conflates units tests with integration tests. The disk driver, for example, requires running a separate read endpoint, which you don't want to have to spin up to run unit tests. The drivers should be tested to ensure that they uphold the driver model separately --- the rest of the code should be able to use a mock driver (i.e., the in memory driver). Integration tests, however, shouldn't use the in memory driver.

> If this is only for testing, then why have the in-memory driver in the main source tree, instead of within the test itself?

It's in the test source directories, and could be consumed by multiple tests, it seems fine to pull it into its own test file.

Reply:
Gotcha.  Thanks for the clarification :+1: